Tests/StoreKitUnitTests/PurchasesOrchestratorSK2Tests.swift

```diff
@@ -726,6 +726,27 @@ class PurchasesOrchestratorSK2Tests: BasePurchasesOrchestratorTests, PurchasesOr
         expect(customerInfo) == mockCustomerInfo
     }
 
+    func testSyncPurchasesSK2DoesNotPostReceiptAndRaisesErrorIfNoTransactionsAndNoAppTransactionJWSToken()
+    async throws {
+        self.mockTransactionFetcher.stubbedFirstVerifiedTransaction = nil
+        self.mockTransactionFetcher.stubbedAppTransactionJWS = nil
+        self.customerInfoManager.stubbedCachedCustomerInfoResult = CustomerInfo.missingOriginalApplicationVersion
+
+        do {
+            _ = try await self.orchestrator.syncPurchases(receiptRefreshPolicy: .always,
+                                                          isRestore: true,
+                                                          initiationSource: .restore)
+            fail("Expected error")
+        } catch {
+            expect(error).to(matchError(ErrorCode.storeProblemError))
```

Comment:
Is there a way to make sure that this error is thrown from the main thread?

Reply:
not here afaik, we'll figure out a different way to test this

Reply:
I was just doing some similar stuff on the retry PR, one thing that you might be able to do to help is to mock out the `OperationDispatcher` and then verify that `invokedDispatchOnMainThread` is true. While you're not directly testing the thread, it's close in that it's verifying how the OperationDispatcher is executing the work.